pymatgen/entries/corrections_calc.py

```diff
@@ -0,0 +1,433 @@
+        calc_compounds: dictionary of ComputedEntry objects
+        corrections: list of corrections in same order as species list
+        corrections_std_error: list of the variances of the corrections in same order as species list
+        corrections_dict: dictionary of format {'species': (value, uncertainty)} for easier correction lookup
+    """
+
+    species = [
+        "oxide",
+        "peroxide",
+        "superoxide",
+        "sulfide",
+        "F",
+        "Cl",
+        "Br",
+        "I",
+        "N",
+        "Se",
+        "Si",
+        "Sb",
+        "Te",
+        "V",
+        "Cr",
+        "Mn",
+        "Fe",
+        "Co",
+        "Ni",
+        "W",
+        "Mo",
+        "H",
+    ]  # species that we're fitting corrections for
+
+    def __init__(self, exp_gz: str, comp_gz: str) -> None:
+
+        """
+        Initializes a CorrectionCalculator.
+
+        Args:
+            exp_gz: name of gzip file that contains experimental data
+                    data in gzip file should be a list of dictionary objects with the following keys/values:
+                    {"formula": chemical formula, "exp energy": formation energy in eV/formula unit,
+                    "uncertainty": uncertainty in formation energy, "warnings": {"polyanion": str describing
+                    polyanions in compound or None, "large_uncertainty": str describing if compound has a large
+                    uncertainty value or None, "unstable": str describing if compound has high e above hull or None}}
+            comp_gz: name of gzip file that contains computed entries
+                    data in gzip file should be a dictionary of {chemical formula: ComputedEntry}
+        """
+
+        self.exp_compounds = loadfn(exp_gz)  # experimental data
+        self.calc_compounds = loadfn(comp_gz)  # computed entries
+        self.corrections: List[float] = []
+        self.corrections_std_error: List[float] = []
+        self.corrections_dict: Dict[
+            str, Tuple[float, float]
+        ] = {}  # {'species': (value, uncertainty)}
+
+        # to help the graph_residual_error_per_species() method differentiate between oxygen containing compounds
+        self.oxides: List[str] = []
+        self.peroxides: List[str] = []
+        self.superoxides: List[str] = []
+        self.sulfides: List[str] = []
+
+    def compute_corrections(
+        self,
+        allow_polyanions: bool = False,
+        allow_large_errors: bool = False,
+        allow_unstable: bool = False,
+    ) -> dict:
+
+        """
+        Computes the corrections and fills in correction, corrections_std_error, and corrections_dict.
+
+        Args:
+            allow_polyanions: optional variable, boolean controlling whether compounds with problematic polyanions
+                will be included in the fit
+            allow_large_errors: optional variable, boolean controlling whether compounds with large experimental
```

Comment:
You're correct that the definitions are hard-coded in the .json file. That is, they are pulled from a 'warnings' key associated with each `ComputedEntry` in the .json file. For this work I believe @awvio defined 'large error' as a relative uncertainty of 10% or greater.

It is true that we could compute the relative uncertainty (indicated by the 'large errors' flag) on the fly and therefore specify a cutoff value in `compute_corrections`, whereas for the other two warning categories - polyanions and e above hull, it's more robust to specify the warnings directly. I think we were trying to maintain consistency with the other two and hence placed the large errors warning in the .json as well.

These warnings are applied to the experimental, not the Computed, compound entries. Hence, it would be difficult to determine `allow_unstable` in code, because that would require matching each experimental entry with a `ComputedEntry` having an `e_above_hull` attribute, which we can of course assume for MP entries, but at the risk of making the calculator less general.

@shyamd @awvio, what are your thoughts about making the warnings / excluded compound filters more automatic or fine-grained?

Reply:
I would argue that these options should actually go into the `__init__` and the `calculate` function takes the two data sets as args.

It would also be good to encapsulate the file loading by making the main calculate take two lists of entries and having helper function that loads from GZip files/